cargo/workspace.bzl

```diff
@@ -0,0 +1,251 @@
+    if repository_ctx.os.environ.get("RULES_RUST_UPDATE_CRATE_UNIVERSE_LOCKFILE", "false") == "true":
+        args.append("--update-lockfile")
+
+    path_binaries = [
+        repository_ctx.path(rustc_label),
+    ]
+
+    result = repository_ctx.execute(
+        args,
+        environment = {
+            "RUST_LOG": "info",
+
+            # The resolver invokes `cargo metadata` which relies on `rustc` being on the $PATH
+            "PATH": ":".join([str(b.dirname) for b in path_binaries]),
+        },
+    )
+    repository_ctx.delete(input_path)
+    if result.stderr:
+        print("Output from resolver: " + result.stderr)
+    if result.return_code != 0:
+        fail("Error resolving deps:\n" + result.stdout + "\n" + result.stderr)
+
+    repository_ctx.file("BUILD.bazel")
+
+_crate_universe_resolve = repository_rule(
+    implementation = _crate_universe_resolve_impl,
+    attrs = {
+        "packages": attr.string_list(allow_empty = True),
+        "cargo_toml_files": attr.label_list(),
+        "overrides": attr.string_dict(doc = """Mapping of crate name to crate.override(...) entries)
+
+Example:
+
+    load("@rules_rust//cargo:workspace.bzl", "rust_library", "crate")
+
+    rust_library(
+        name = "mylib",
+        # [...]
+        overrides = {
+            "tokio": crate.override(
+                extra_rust_env_vars = {
+                    "MY_ENV_VAR": "MY_ENV_VALUE",
+                },
+                extra_build_script_env_vars = {
+                    "MY_BUILD_SCRIPT_ENV_VAR": "MY_ENV_VALUE",
+                },
+                extra_bazel_deps = {
+                    # Extra dependencies are per target. They are additive.
+                    "cfg(unix)": ["@somerepo//:foo"],  # cfg() predicate.
+                    "x86_64-apple-darwin": ["@somerepo//:bar"],  # Specific triple.
+                    "cfg(all())": ["@somerepo//:baz"],  # Applies to all targets ("regular dependency").
+                },
+                extra_build_script_bazel_deps = {
+                    # Extra dependencies are per target. They are additive.
+                    "cfg(unix)": ["@buildscriptdep//:foo"],
+                    "x86_64-apple-darwin": ["@buildscriptdep//:bar"],
+                    "cfg(all())": ["@buildscriptdep//:baz"],
+                },
+                extra_bazel_data_deps = {
+                    # ...
+                },
+                extra_build_script_bazel_data_deps = {
+                    # ...
+                },
+            ),
+        },
+    )
+
+"""),
+        "repository_template": attr.string(),
+        "supported_targets": attr.string_list(allow_empty = False),
+        "lockfile": attr.label(
+            allow_single_file = True,
+            mandatory = False,
+        ),
+        "_resolver_script_linux": attr.label(
+            default = "@crate_universe_resolver_linux//file:downloaded",
+            executable = True,
+            cfg = "host",
```

Comment:
I think these should be `exec`

Reply:
Repository rules run on the host machine where bazel is running, not on an exec platform